### PR TITLE
Global schema

### PR DESCRIPTION
**Pull request by lolopinto**
This adds a global schema which is useful for a couple things right now:

* adds global edges https://github.com/lolopinto/ent/issues/744
 (not fully fleshed out, follow-up task https://github.com/lolopinto/ent/issues/1032)
* adds global configuration for edges e.g. add extra fields to the edge table, do soft deleting of edges etc similar to what we do for writes on a per-schema basis (usually in a pattern if we want it everywhere) https://github.com/lolopinto/ent/pull/830

### CODE REVIEW

File: ts/src/core/ent.ts [hunk 1]
```diff
@@ -1332,37 +1420,30 @@ export function createRowSync(
...
     if (options.fieldsToLog) {
       logValues.push(options.fieldsToLog[key]);
     }
-    if (dialect === Dialect.Postgres) {
-      valsString.push(`${key} = $${idx}`);
-      idx++;
-    } else {
-      valsString.push(`${key} = ?`);
-    }
+    valsString.push(clause.Eq(key, val).clause(idx));
```
Comment (lolopinto) [inline ts/src/core/ent.ts:1436]: TODO this is wrong.

leads to `foo IS NULL` queries when set to null

TODO add test for setting to null correctly
